### PR TITLE
Allow ECR push/pull with URI

### PR DESCRIPTION
**Pull request by yinshiua**
```
$ make build
$ make generate
$ make test
PASS
```

```
$ ./bin/local/ecs-cli pull ********.dkr.ecr.us-east-1.amazonaws.com/busybox
INFO[0000] Pulling image                                 repository="**********.dkr.ecr.us-east-1.amazonaws.com/busybox" tag=
INFO[0002] Image pulled                                 

$ ./bin/local/ecs-cli push ***********.dkr.ecr.us-east-1.amazonaws.com/busybox  
INFO[0000] Pushing image                                 repository="******.dkr.ecr.us-east-1.amazonaws.com/busybox" tag=
INFO[0005] Image pushed

$  ./bin/local/ecs-cli push busybox
INFO[0000] Getting AWS account ID...                    
INFO[0000] Tagging image                                 image=busybox repository="********.dkr.ecr.us-east-1.amazonaws.com/busybox" tag=
INFO[0000] Image tagged                                 
INFO[0000] Pushing image                                 repository="********.dkr.ecr.us-east-1.amazonaws.com/busybox" tag=
INFO[0007] Image pushed 

                                
$ ./bin/local/ecs-cli push busybox:latest
INFO[0000] Getting AWS account ID...                    
INFO[0000] Tagging image                                 image="busybox:latest" repository="********.dkr.ecr.us-east-1.amazonaws.com/busybox" tag=latest
INFO[0000] Image tagged                                 
INFO[0000] Pushing image                                 repository="********.dkr.ecr.us-east-1.amazonaws.com/busybox" tag=latest
INFO[0002] Image pushed  
```

### CODE REVIEW

File: ecs-cli/modules/command/image_app.go [hunk 1]
```diff
@@ -324,15 +294,28 @@ func getRegistryID(registryID string, stsClient stsclient.Client) (string, error
...
 	}
-	repository := s[0]
-	tag := ""
-	if len(s) == 2 {
-		tag = s[1]
+
+	return ecrClient.GetAuthorizationToken(registryURI)
+}
+
+func splitImageName(image string, seperatorRegExp string,
```
Comment (aaithal) [inline ecs-cli/modules/command/image_app.go]: This method warrants it's own unit test if one doesn't exist yet.

File: ecs-cli/modules/command/image_app.go [hunk 2]
```diff
@@ -124,51 +123,31 @@ func ImageList(c *cli.Context) {
...
-	}
-
-	repository, tag, err := splitImageName(targetImage, SeperatorColon, PUSH_IMAGE_FORMAT)
-	if err != nil {
-		return err
-	}
+	image := args[0]
 
-	registryID, err = getRegistryID(registryID, stsClient)
+	registryURI, repository, tag, err := splitImageName(image, "[:]", PushImageFormat)
```
Comment (aaithal) [inline ecs-cli/modules/command/image_app.go]: Do we not support `@` with ECR?

Comment (yinshiua) [inline ecs-cli/modules/command/image_app.go]: docker push doesn't allow push image with sha256

File: ecs-cli/modules/command/image_app.go (hunk 1 above)
Comment (yinshiua) [inline ecs-cli/modules/command/image_app.go]: added tests

**Pull request merged**